packages/connectors-web/src/alchemySmartWallet.ts

```diff
@@ -1,85 +1,233 @@
+        }
+        const baseProvider = (await innerConnector.getProvider()) as any;
+        return baseProvider.request(requestParams);
+      },
+    };
+
+    let smartWalletClientPromise: Promise<SmartWalletClient> | undefined;
+
+    const resetState = (): void => {
+      smartWalletClientPromise = undefined;
+    };
+
+    const createClient = async (): Promise<SmartWalletClient> => {
+      const chainId = await innerConnector.getChainId();
+      const innerAccounts = await innerConnector.getAccounts();
+      if (innerAccounts.length === 0) {
+        throw new Error("No accounts found in base connector");
+      }
+      // TODO: do we always take the first account?
+      const signerAddress = innerAccounts[0];
+      const chain = getChainFromConfig(config, chainId);
+      const provider = (await innerConnector.getProvider()) as EIP1193Provider;
+      const signer = createWalletClient({
+        account: signerAddress,
+        chain,
+        transport: custom(provider),
+      });
+      const smartWalletClient = createSmartWalletClient({
+        signer,
+        transport: alchemyTransport({
+          apiKey: options.apiKey,
+          jwt: options.jwt,
+          url: options.url ?? "https://api.g.alchemy.com/v2",
+        }),
+        chain,
+      });
+      // TODO: Calling `client.getProvider()` has the side effect of
+      // initializing `client.account` some time after the provider is created.
+      // We should change that so it's explicit and awaitable. Until then,
+      // replicate the logic by hand.
+      const account = await smartWalletClient.requestAccount();
+      smartWalletClient.account = {
+        type: "json-rpc",
+        address: account.address,
+      };
+      return smartWalletClient;
+    };
+
+    const initializeSmartWalletClient = async (): Promise<void> => {
+      smartWalletClientPromise = createClient();
+      await smartWalletClientPromise;
+    };
+
+    // A trivial function, but gives a name to indicate how the promise is used.
+    const getSmartWalletClientIfConnected = async (): Promise<
+      SmartWalletClient | undefined
+    > => {
+      return smartWalletClientPromise;
+    };
 
     return {
       id: "alchemySmartWallet",
       name: "Alchemy Smart Wallet",
       type: alchemySmartWallet.type,
-      baseConnector,
+      baseConnector: innerConnector,
 
       async setup(): Promise<void> {
-        await baseConnector.setup?.();
-
-        // Optional function for running when the connector is first created.
-        throw new Error("Not implemented: setup");
+        await innerConnector.setup?.();
+        if (await innerConnector.isAuthorized()) {
+          await initializeSmartWalletClient();
+        }
       },
 
-      async connect(parameters) {
-        await baseConnector.connect(parameters);
-        throw new Error("Not implemented: connect");
+      async connect(params): Promise<{ accounts: Address[]; chainId: number }> {
```

Comment:
not sure if this would change anything for your implementation, but just want to flag that `connect` has an `{ isReconnecting: true }` param whenever it's called via wagmi's `reconnect` action.